distributed/utils.py

```diff
@@ -1841,52 +1841,73 @@ def is_python_shutting_down() -> bool:
 class Deadline:
     """Utility class tracking a deadline and the progress toward it"""
 
-    #: Expiry time of the deadline in seconds since the epoch
+    #: Expiry time of the deadline in seconds since the start of the monotonic timer
     #: or None if the deadline never expires
-    expires_at: float | None
+    expires_at_mono: float | None
     #: Seconds since the epoch when the deadline was created
+    started_at_mono: float
+    #: Seconds since the start of the monotonic timer when the deadline was created
     started_at: float
 
+    __slots__ = tuple(__annotations__)
+
     def __init__(self, expires_at: float | None = None):
-        self.expires_at = expires_at
         self.started_at = time()
+        self.started_at_mono = monotonic()
+        if expires_at is not None:
+            self.expires_at_mono = expires_at - self.started_at + self.started_at_mono
+        else:
+            self.expires_at_mono = None
 
     @classmethod
     def after(cls, duration: float | None = None) -> Deadline:
         """Create a new ``Deadline`` that expires in ``duration`` seconds
-        or never if ``duration`` is None"""
-        started_at = time()
-        expires_at = duration + started_at if duration is not None else duration
-        deadline = cls(expires_at)
-        deadline.started_at = started_at
-        return deadline
+        or never if ``duration`` is None
+        """
+        inst = cls()
+        if duration is not None:
+            inst.expires_at_mono = inst.started_at_mono + duration
+        return inst
+
+    @property
+    def expires_at(self) -> float | None:
```

Comment:
I'm not even sure if this is strictly necessary or if we should just remove it and make `expires_at_mono` private.